src/internal.c

```diff
@@ -16110,9 +16110,31 @@ int ProcessReply(WOLFSSL* ssl)
                                       in->buffer + in->idx,
                                       ssl->curSize - (word16)digestSz);
                         if (ret == 0) {
-                            ssl->keys.padSz =
-                                    in->buffer[in->idx + ssl->curSize -
-                                               digestSz - 1];
+                            byte invalid = 0;
+                            byte padding = (byte)-1;
+                            word32 i;
+                            word32 off = in->idx + ssl->curSize - digestSz - 1;
```

Comment:
I realize this is existing logic, but what assurances do we have this will result in a positive value? This is unsigned... I checked the code above and not seeing explicit checks for these variables. Can we improve this for safety?

Reply:
SanityCheckCipherText() checks ssl->curSize is big enough.